documents/entityrelationship/model.go

```diff
@@ -0,0 +1,319 @@
+type EntityRelationship struct {
+	*documents.CoreDocument
+
+	// owner of the relationship
+	OwnerIdentity *identity.DID
+	// document identifier
+	Label []byte
+	// identity which will be granted access
+	TargetIdentity *identity.DID
+}
+
+// getClientData returns the entity relationship data from the entity relationship model
+func (e *EntityRelationship) getClientData() *cliententitypb.EntityRelationshipData {
+	var owner string
+	var target string
+	label := hexutil.Encode(e.Label)
+	if e.OwnerIdentity != nil {
+		owner = e.OwnerIdentity.String()
+	}
+	if e.TargetIdentity != nil {
+		target = e.TargetIdentity.String()
+	}
+
+	return &cliententitypb.EntityRelationshipData{
+		OwnerIdentity:  owner,
+		Label:          label,
+		TargetIdentity: target,
+	}
+}
+
+// createP2PProtobuf returns Centrifuge protobuf-specific EntityRelationshipData.
+func (e *EntityRelationship) createP2PProtobuf() *entitypb.EntityRelationship {
+	var didByte []byte
+	var tidByte []byte
+	if e.OwnerIdentity != nil {
+		didByte = e.OwnerIdentity[:]
+	}
+	if e.TargetIdentity != nil {
+		tidByte = e.TargetIdentity[:]
+	}
+
+	return &entitypb.EntityRelationship{
+		OwnerIdentity:  didByte,
+		Label:          e.Label,
+		TargetIdentity: tidByte,
+	}
+}
+
+// InitEntityRelationshipInput initialize the model based on the received parameters from the rest api call
+func (e *EntityRelationship) InitEntityRelationshipInput(payload *cliententitypb.EntityRelationshipCreatePayload) error {
+	if err := e.initEntityRelationshipFromData(payload.Data); err != nil {
+		return err
+	}
+
+	cd, err := documents.NewCoreDocumentWithCollaborators([]string{payload.Data.OwnerIdentity}, compactPrefix())
+	if err != nil {
+		return errors.New("failed to init core document: %v", err)
+	}
+
+	e.CoreDocument = cd
+	return nil
+}
+
+// PrepareNewVersion prepares new version from the old entity.
+func (e *EntityRelationship) PrepareNewVersion(old documents.Model, data *cliententitypb.EntityRelationshipData, collaborators []string) error {
+	err := e.initEntityRelationshipFromData(data)
+	if err != nil {
+		return err
+	}
+
+	oldCD := old.(*EntityRelationship).CoreDocument
+	e.CoreDocument, err = oldCD.PrepareNewVersion(compactPrefix())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// initEntityRelationshipFromData initialises an EntityRelationship from entityRelationshipData.
+func (e *EntityRelationship) initEntityRelationshipFromData(data *cliententitypb.EntityRelationshipData) error {
+	if data.OwnerIdentity != "" {
+		if did, err := identity.NewDIDFromString(data.OwnerIdentity); err == nil {
+			e.OwnerIdentity = &did
+		} else {
+			return err
+		}
+	} else {
+		return identity.ErrMalformedAddress
+	}
+
+	if data.TargetIdentity != "" {
+		if did, err := identity.NewDIDFromString(data.TargetIdentity); err == nil {
+			e.TargetIdentity = &did
+		} else {
+			return err
+		}
+	} else {
+		return identity.ErrMalformedAddress
+	}
+	e.Label = []byte(data.Label)
```

Comment:
We need to use here: 
```suggestion
	e.Label, err = hexutil.Decode(data.Label)
```